workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs

```diff
@@ -476,5 +450,94 @@ private static IEnumerable<string> FindDirInPackages(string directory)
+        {
+            toolsConfig = toolsConfig ?? GdkToolsConfiguration.GetOrCreateInstance();
+
+            var schemaCompilerPath = GetSchemaCompilerPath();
+            var logfilePath = GetDefaultLoggerOutputPath();
+
+            var codegenArgs = new List<string>
+            {
+                $"--json-dir=\"{ImprobableJsonDir}\"",
+                $"--schema-compiler-path=\"{schemaCompilerPath}\"",
+                $"--worker-json-dir=\"{WorkerJsonPath}\"",
+                $"--log-file=\"{logfilePath}\""
+            };
+
+            if (toolsConfig.VerboseLogging)
+            {
+                codegenArgs.Add($"--verbose");
+            }
+
+            var codegenOutputPath = Path.GetFullPath(Path.Combine(Application.dataPath, "..", toolsConfig.CodegenOutputDir));
+            codegenArgs.Add($"--native-output-dir=\"{codegenOutputPath}\"");
+
+            // Add user defined schema directories, warn if directory does not exist
+            foreach (var schemaDir in toolsConfig.SchemaSourceDirs)
+            {
+                if (!Directory.Exists(schemaDir))
+                {
+                    Debug.LogWarning($"Schema directory defined in GDK tools configuration does not exist: {schemaDir}");
+                    continue;
+                }
+
+                codegenArgs.Add($"--schema-path=\"{Path.GetFullPath(schemaDir)}\"");
+            }
+
+            // Add package schema directories
+            codegenArgs.AddRange(FindDirInPackages(SchemaPackageDir)
+                .Select(directory => $"--schema-path=\"{directory}\""));
+
+            // Schema Descriptor
+            codegenArgs.Add($"--descriptor-dir=\"{toolsConfig.DescriptorOutputDir}\"");
+
+            codegenArgs.AddRange(
+                toolsConfig.SerializationOverrides.Select(@override => $"--serialization-override=\"{@override}\""));
+
+            var codegenArgsString = string.Join(" ", codegenArgs);
+
+            // For dotnet run / visual studio
+            try
+            {
+                var csprojXml = XDocument.Load(CodegenExe);
+                var projectNode = csprojXml.Element("Project");
+                var propertyGroup = projectNode.Element("PropertyGroup");
+
+                var args = propertyGroup.Element("StartArguments");
+                args?.Remove();
+
+                propertyGroup.Add(XElement.Parse($"<StartArguments>{codegenArgsString}</StartArguments>"));
+                csprojXml.Save(CodegenExe);
+            }
+            catch
+            {
+                throw new Exception("Unable to update csproj with run configuration.");
+            }
+
+            // For jetbrains rider
+            try
+            {
+                var runConfigPath = Path.Combine(CodegenExeDirectory, ".idea", ".idea.CodeGen", ".idea",
+                    "runConfigurations");
+                Directory.CreateDirectory(runConfigPath);
```

Comment:
What is this directory already exists with files in it?

Reply:
this'll do nothing

Q: but do we not need to wipe the contents of the dir?
A: we only care about the run config xml and we're overwriting that anyway

Q: but we remove the args in the csproj?
A: sadly there's no nice way of easily overwriting the csproj node

Q: ok